torch/_inductor/lowering.py

```diff
@@ -2908,6 +2908,7 @@ def inner_fn(idx):
 
 @register_lowering(aten.slice_scatter, type_promotion_kind=None)
 def slice_scatter(x, src, dim=0, start=None, end=None, step=1):
+    src = to_dtype(src, x.get_dtype())
```

Comment:
I was hoping setting type_promotion_kind can do the right type promotion, but looks like we do need to convert to x's dtype here to match the eager behavior.  cc @eellison